swap/htlc.go

```diff
@@ -207,6 +223,50 @@ func swapHTLCScriptV1(cltvExpiry int32, senderHtlcKey,
 	return builder.Script()
 }
 
+// swapHTLCScriptV2 returns the on-chain HTLC V2 witness script.
+//
+// <receiverHtlcKey> OP_CHECKSIG OP_NOTIF
+//   OP_DUP OP_HASH160 <HASH160(senderHtlcKey)> OP_EQUALVERIFY OP_CHECKSIGVERIFY
+//   <cltv timeout> OP_CHECKLOCKTIMEVERIFY
+// OP_ELSE
+//   OP_SIZE <20> OP_EQUALVERIFY OP_HASH160 <ripemd(swapHash)> OP_EQUALVERIFY 1
+//   OP_CHECKSEQUENCEVERIFY
+// OP_ENDIF
+func swapHTLCScriptV2(cltvExpiry int32, senderHtlcKey,
+	receiverHtlcKey [33]byte, swapHash lntypes.Hash) ([]byte, error) {
+
+	builder := txscript.NewScriptBuilder()
+	builder.AddData(receiverHtlcKey[:])
+	builder.AddOp(txscript.OP_CHECKSIG)
+
+	builder.AddOp(txscript.OP_NOTIF)
+
+	builder.AddOp(txscript.OP_DUP)
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(input.Ripemd160H(senderHtlcKey[:]))
+	builder.AddOp(txscript.OP_EQUALVERIFY)
+	builder.AddOp(txscript.OP_CHECKSIGVERIFY)
+
+	builder.AddInt64(int64(cltvExpiry))
+	builder.AddOp(txscript.OP_CHECKLOCKTIMEVERIFY)
+
+	builder.AddOp(txscript.OP_ELSE)
+
+	builder.AddOp(txscript.OP_SIZE)
+	builder.AddInt64(20)
+	builder.AddOp(txscript.OP_EQUALVERIFY)
+	builder.AddOp(txscript.OP_HASH160)
+	builder.AddData(input.Ripemd160H(swapHash[:]))
+	builder.AddOp(txscript.OP_EQUALVERIFY)
+	builder.AddInt64(1)
```

Comment:
Hmm maybe this is better as `OP_1`? I am no expert.

Reply:
I'm not expert either, but looking at `ScriptBuilder` it is equivalent.

Reply:
Also produce the same byte code when compiled with `btcc`.

Reply:
The original scripts as specified in https://github.com/lightninglabs/nautilus/issues/222 uses `1` and not `<1>`

Reply:
Discussed this offline: changed to `OP_1` to conform with coding style elsewhere.

Reply:
But it isn't `OP_1` right now?